### PR TITLE
Newsletters: add 305 (2024-05-31)

### PR DESCRIPTION
**Pull request by harding**
- [ ] Lede, releases/RCs, topic entries @harding 
- [x] PR summaries @Gustavojfe 
- [x] Stack Exchange @bitschmidty 

Note: appologies, this draft might be a bit rough; I was a bit rushed for time and my preview setup broke.

### CODE REVIEW

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+---
+This week's newsletter FIXME
+
+## News
+
+- **Light client protocol for silent payments:** FIXME [posted][fixme
+  lcsp] to Delving Bitcoin to describe a draft specification for a
+  protocol to help lightweight clients receive [silent payments][topic
+  silent payments] (SPs).  The addition of a few cryptographic
+  primatives is sufficient to give any wallet software the ability to
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-05-31-newsletter.md:18]:
```suggestion
  primitives is sufficient to give any wallet software the ability to
```

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  send SPs, but receiving silent payments requires not only those
+  primatives but also the ability to access information about every
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-05-31-newsletter.md:20]:
```suggestion
  primitives but also the ability to access information about every
```

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  block-level data that allows them to learn how much they've received
+  and how to later spend the payment.
+
+- **Raw taproot descriptors:** FIXME [posted][fixme descriptors] to
+  Delving Bitcoin about two new proposed [descriptors][topic
+  descriptors] for constructing [taproot][topic taproot] spend
+  conditions:
+
+  - `rawnode(<hash>)` takes the hash of a merkle tree node, whether for an
+    internal node or for leaf node.  This a wallet or other scanning
```
Comment (vostrnad) [inline _posts/en/newsletters/2024-05-31-newsletter.md:41]:
```suggestion
    internal node or for leaf node.  This allows a wallet or other scanning
```

**Comment by bitschmidty** [comment]
Pushed a stack exchange commit, which felt light this month, so definitely open to additions.

**Comment by vostrnad** [comment]
Some highly upvoted Stack Exchange answers that I think are worth adding:

- [Why is my P2P_V2 peer connected over a v1 connection?](https://bitcoin.stackexchange.com/questions/122774/why-is-my-p2p-v2-peer-connected-over-a-v1-connection) (disclaimer: question by me)
- [Does a P2PKH transaction send to the hash of the uncompressed key or the compressed key?](https://bitcoin.stackexchange.com/questions/122875/does-a-p2pkh-transaction-send-to-the-hash-of-the-uncompressed-key-or-the-compres)
- [Broadcast a valid block to network, without a local full node](https://bitcoin.stackexchange.com/questions/122953/broadcast-a-valid-block-to-network-without-a-local-full-node)

**Comment by bitschmidty** [comment]
> Some highly upvoted Stack Exchange answers that I think are worth adding:
> 
> * [Why is my P2P_V2 peer connected over a v1 connection?](https://bitcoin.stackexchange.com/questions/122774/why-is-my-p2p-v2-peer-connected-over-a-v1-connection) (disclaimer: question by me)
> * [Does a P2PKH transaction send to the hash of the uncompressed key or the compressed key?](https://bitcoin.stackexchange.com/questions/122875/does-a-p2pkh-transaction-send-to-the-hash-of-the-uncompressed-key-or-the-compres)
> * [Broadcast a valid block to network, without a local full node](https://bitcoin.stackexchange.com/questions/122953/broadcast-a-valid-block-to-network-without-a-local-full-node)

Great suggestions, added all 3 thank you @vostrnad

**Comment by Gustavojfe** [comment]
just pushed the 6 merged PR summaries for this week. quite interesting stuff. Thanks for the feedback and review.

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    she gives him a templated descriptor (perhaps including a timelock
+    that prevents him from spending until a period of time has past).
+    This is safe for Bob because the money isn't his and it's good for
+    Alice's privacy because she doesn't need to reveal any of her other
+    spend conditions to Bob upfront (although Bob may learn them from
+    Alice's onchain transactions).
+
+  - `rawleaf(<script>,[version])` is similar to the existing `raw`
+    descriptor for including scripts that can't be expressed using a
+    templated descriptor.  It's main difference is that it includes the
```
Comment (LarryRuane) [inline _posts/en/newsletters/2024-05-31-newsletter.md:60]:
```suggestion
    templated descriptor.  Its main difference is that it includes the
```

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+  `inv` / `getdata` / `block` message flow.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND v0.18.0-beta.rc3][] is a release candidate for the next major
+  version of this popular LN node implementation.
```
Comment (jirijakes) [inline _posts/en/newsletters/2024-05-31-newsletter.md:135]: In the meantime, [beta](https://github.com/lightningnetwork/lnd/releases/tag/v0.18.0-beta) was released.

```suggestion
- [LND v0.18.0-beta][] is beta release of the next major
  version of this popular LN node implementation.
```

**Comment by harding** [comment]
Made edits, added lede and releases, and reviewed other contributions.  Didn't add topic entries, will do that post merge.  Thanks everyone!

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+  order to help simulate high feerate environments.
+
+- [Why is my P2P_V2 peer connected over a v1 connection?]({{bse}}122774)
+  Pieter Wuille hypothesizes that outdated peer addr information was what caused
+  a user to see a peer that supported BIP324 [encrypted transport][topic v2 p2p
+  transport] that was connected with a v1 non-encrypted connection.
+
+- [Does a P2PKH transaction send to the hash of the uncompressed key or the compressed key?]({{bse}}122875)
+  Pieter Wuille notes that both compressed and uncompressed public keys can be
+  used resulting in distinct P2PKH addresses, adding that P2PKH only supports
```
Comment (jirijakes) [inline _posts/en/newsletters/2024-05-31-newsletter.md:126]:
```suggestion
  used resulting in distinct P2PKH addresses, adding that P2WPKH only supports
```

Comment (vostrnad) [reply _posts/en/newsletters/2024-05-31-newsletter.md:126]: Also, P2WPKH does support uncompressed keys, they're just non-standard by Bitcoin Core's policy. How about:
> P2WPKH only allows compressed public keys by policy and P2TR uses x-only public keys.

File: _posts/en/newsletters/2024-05-31-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+  [#178][news178 txoutset] and [#72][news72 txoutset] for previous references to
+  `dumptxoutset`.
+
+- [Bitcoin Core #27064][] changes the default data directory on Windows from
+  `C:\Users\Username\AppData\Roaming\Bitcoin` to
+  `C:\Users\Username\AppData\Local\Bitcoin` on fresh installs only.
+
+- [Bitcoin Core #29873][] introduces a 10 kvB data weight limit for
+  [Topologically Restricted Until Confirmation (TRUC)][topic v3 transaction
+  relay] transactions (v3 transactions) to reduce the potential cost of
```
Comment (jirijakes) [inline _posts/en/newsletters/2024-05-31-newsletter.md:177]: Sorry for late review but I think this is quite a significant omission.

The PR makes it cheaper to defend against pinning attacks, however the original text sounds like it makes attacking cheaper.

@glozow says:

> It lowers potential replacement cost (i.e. harder to do Rule 3 pinning via gigantic transaction)

```suggestion
  relay] transactions (v3 transactions) to reduce the potential cost of mitigation against
```

Comment (bitschmidty) [reply _posts/en/newsletters/2024-05-31-newsletter.md:177]: Taking this suggestion

File: _posts/en/newsletters/2024-05-31-newsletter.md (hunk 6 above)
Comment (bitschmidty) [reply _posts/en/newsletters/2024-05-31-newsletter.md:126]: Taking this suggestion, thanks @jirijakes and @vostrnad

File: _posts/en/newsletters/2024-05-31-newsletter.md (hunk 4 above)
Comment (bitschmidty) [reply _posts/en/newsletters/2024-05-31-newsletter.md:66]: Taking this suggestion

**Comment by bitschmidty** [comment]
Took suggestions noted above, made a fixup to the lede, squashed and merged, thank you @harding @Gustavojfe @jirijakes @vostrnad @LarryRuane for your contributions this week! 🚀

**Pull request merged**